src-tauri/src/libs/utils.rs

```diff
@@ -28,12 +34,54 @@ impl TimeLogger {
     }
 }
 
-/**
- * Get the app configuration/storage directory
- */
-pub fn get_app_storage_dir() -> PathBuf {
-    let path = dirs::home_dir().expect("Get home dir");
-    path.join(".museeks")
+#[must_use]
+const fn app_name() -> &'static str {
+    // Must match the value of "identifier" in tauri.config.json!
+    "io.museeks.app"
+}
+
+#[must_use]
+fn app_dirs() -> Option<ProjectDirs> {
+    ProjectDirs::from("", "", app_name())
+}
+
+fn init_app_dir(app_dir: &Path) {
+    if let Err(err) = fs::create_dir_all(app_dir) {
+        log::error!(
+            "Failed to create app directory '{dir}': {err}",
+            dir = app_dir.display(),
+        );
+    }
+}
+
+#[must_use]
+fn init_config_dir(app_dirs: &ProjectDirs) -> &Path {
```

Comment:
imho, all config related code should go to the `config.rs` for the sake of not jumping between 5 files when reading the config code.